libs/ledger/tests/unit/storage/transaction_archiver_tests.cpp

```diff
@@ -0,0 +1,141 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "ledger/storage_unit/transaction_archiver.hpp"
+#include "mock_transaction_pool.hpp"
+#include "mock_transaction_store.hpp"
+#include "transaction_generator.hpp"
+
+#include "gtest/gtest.h"
+
+namespace {
+
+using testing::_;
+using testing::InSequence;
+using testing::Return;
+using testing::NiceMock;
+using fetch::ledger::TransactionArchiver;
+
+class TransactionArchiverTests : public ::testing::Test
+{
+protected:
+  void CycleStateMachine()
+  {
+    auto state_machine = archiver_.GetStateMachine();
+
+    for (std::size_t i = 0; i < 10; ++i)
+    {
+      state_machine->Execute();
+
+      // exit the state machine when the next state is collecting again
+      if (state_machine->state() == TransactionArchiver::State::COLLECTING)
+      {
+        return;
+      }
+    }
+
+    FAIL() << "State machine never completed";
+  }
+
+  NiceMock<MockTransactionPool>  pool_;
+  NiceMock<MockTransactionStore> store_;
+  TransactionGenerator           tx_gen_;
+  TransactionArchiver            archiver_{pool_, store_};
+};
+
+MATCHER_P(IsTransaction, digest, "")  // NOLINT
```

Comment:
What happens if you don't nolint this?

Reply:
Sorry I can't remember, but if you look all `MATCHER` instances in the code base as it stands are `NOLINT`